File: python/205_同构字符串_easy.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        strMaps, strMapt = dict(), dict()
        for i in range(0, len(s)):
            x, y = s[i], t[i]
            if (x in strMaps and strMaps[x] != y) or (y in strMapt and strMapt[y] != x):
                return False
            strMaps[x] = y
            strMapt[y] = x
        return True

File: cpp/205_同构字符串_easy.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char, char> strMaps;
        unordered_map<char, char> strMapt;
        int N = s.length();
        for(int i=0; i<N; i++){
            char x = s[i], y = t[i];
            if((strMaps.count(x) && strMaps[x]!=y) || strMapt.count(y) && strMapt[y] != x)
                return false;
            strMaps[x] = y;
            strMapt[y] = x;
        }
        return true;
    }
};